packages/api/src/external/carequality/document/process-outbound-document-retrieval-resps.ts

```diff
@@ -89,14 +90,20 @@ export async function processOutboundDocumentRetrievalResps({
       source: MedicalDataSource.CAREQUALITY,
     });
 
+    const seenMetriportIds = new Set<string>();
+
     const resultPromises = await Promise.allSettled(
       results.map(async docRetrievalResp => {
         const docRefs = docRetrievalResp.documentReference;
 
         if (docRefs) {
           const validDocRefs = docRefs.filter(containsMetriportId);
+          const deduplicatedDocRefs = validDocRefs.filter(
+            docRef => !containsDuplicateMetriportId(docRef, seenMetriportIds)
+          );
+
```

Comment:
If we have dups on DR it should be flagged as a programming error, since we already filter on DQ, no?

Reply:
Theoretically yes but there might be edge cases we havent thought through yet. I added a capture here so we will get flagged if this happens and can investigate